request.go

```diff
@@ -294,8 +289,12 @@ func unmarshalNode(data *Node, model reflect.Value, included *map[string]*Node)
 				} else if v.Kind() == reflect.Int {
 					at = v.Int()
 				} else {
-					er = ErrInvalidTime
-					break
+					// Return error immediately to ensure a runtime panic doesn't swallow it.
+					return &ErrorObject{
```

Comment:
So have `requests.go` continue to return the errors that it had previously been returning across the board? Ok. The original reason that I opened this PR is because the actual errors themselves (like the one on this line) were being swallowed by an abstraction and the actual underlying issue was being hidden during unmarshal calls where bad types were present.

If you look at [some previous review which had been given](https://github.com/google/jsonapi/pull/38#issuecomment-240530075), I was asked to do the opposite of what is being asked here. But that is ok. I'll update according to your current request.